### PR TITLE
Add Json Scalar

### LINKED ISSUE #3661: Add Json Scalar
Add `JsonType`  to `Hotchocolate.Types`

Addresses #2927

### PR DESCRIPTION
**Pull request by pedro-coelho-ff**
Add JsonType to Hotchocolate.Types

Closes #3661

### CODE REVIEW

**Comment by wonbyte** [comment]
@pedro-coelho-ff The scalar needs to only support a `JsonDocument` type.

**Comment by michaelstaib** [comment]
is this one ready for review?

**Comment by pedro-coelho-ff** [comment]
> is this one ready for review?

Yes, I believe it is 😅

File: src/HotChocolate/Core/test/Types.Scalars.Tests/JsonTypeTests.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+using System.Text.Json;
+using HotChocolate.Language;
+using HotChocolate.Tests;
+using Snapshooter.Xunit;
+using Xunit;
+
+namespace HotChocolate.Types
+{
+
+
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Types.Scalars.Tests/JsonTypeTests.cs:11]:
```suggestion
```

**Comment by michaelstaib** [comment]
on it.

File: src/HotChocolate/Core/src/Types.Scalars/JsonType.cs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            if (runtimeValue == null)
+            {
+                resultValue = null;
+                return true;
+            }
+
+            return Convert(runtimeValue, out resultValue);
+        }
+
+        public override bool TryDeserialize(object? resultValue, out object? runtimeValue)
```
Comment (michaelstaib) [inline src/HotChocolate/Core/src/Types.Scalars/JsonType.cs:151]: TrySerialize: in object / out JsonDocument

**Review by michaelstaib — CHANGES_REQUESTED** [review]

**Comment by michaelstaib** [comment]
I think I will take a swing at this so we get it into the release.

**Comment by pedro-coelho-ff** [comment]
> I think I will take a swing at this so we get it into the release.

I really appreciate it if you or any other that is more comfortable than me taking this one and putting it in the next release. 😅

**Comment by wonbyte** [comment]
@michaelstaib let me know if you want me to take it

**Comment by pedro-coelho-ff** [comment]
@wonbyte and @michaelstaib sorry to bother you with this pull request.
Just a quick note, I had some performance issues with this approach, the solution was to use the AnyType but instead of JsonObject we created a custom JsonConverter and convert it to an object.

**Pull request closed without merging**